boot/boot_serial/src/boot_serial.c

```diff
@@ -414,6 +413,14 @@ bs_upload(char *buf, int len)
         goto out_invalid_data;
     }
 
+    /* According to the specification, only requests with 'off' set to '0' can
+     * contain image number. Use image number value provided only in first SMP
+     * upload request packet.
+     */
```

Comment:
You do not have to mention doc here, just state that image number is only taken from packets with offset == 0.
If we would mention documentation, in comments, as a reason to every change, the code would get really wired. Unless documentation requires something that is counter-intuitive, you do not have to put justification citing documentation in comments.

Reply:
@de-nordic makes sense. I fixed that and rebased on latest 'main'.

Thanks for your comment!